app/scripts/controller-init/snaps/snaps-registry-init.ts

```diff
@@ -0,0 +1,31 @@
+import { JsonSnapsRegistry } from '@metamask/snaps-controllers';
+import { ControllerInitFunction } from '../types';
+import { SnapsRegistryMessenger } from '../messengers/snaps';
+
+/**
+ * Initialize the Snaps registry controller.
+ *
+ * @param request - The request object.
+ * @param request.controllerMessenger - The messenger to use for the controller.
+ * @param request.persistedState - The persisted state of the extension.
+ * @returns The initialized controller.
+ */
+export const SnapsRegistryInit: ControllerInitFunction<
+  JsonSnapsRegistry,
+  SnapsRegistryMessenger
+> = ({ controllerMessenger, persistedState }) => {
+  const requireAllowlist = Boolean(process.env.REQUIRE_SNAPS_ALLOWLIST);
+
+  const controller = new JsonSnapsRegistry({
+    // @ts-expect-error: `persistedState.SnapsRegistry` is not compatible
+    // with the expected type.
+    // TODO: Look into the type mismatch.
```

Comment:
Asking this again, any idea why all of these types are mismatched?

Reply:
`persistedState` is `Partial`, but the controllers expect a full state object. We will need to address this in a Snaps controllers update. This essentially matches the behaviour of the previous JS implementation, so I think we can do that in a future update.